### ISSUE TITLE
Support for nested secrets

### ISSUE THREAD

**Issue by rajdip-b** [keyshade-xyz/keyshade#87]
## Description

There are multiple scenarios when we would like to group our secrets together, let's say, we would like to group our Google OAuth secrets into a JSON schema:

```json
{
  "google": {
      "accessKey": "foo",
      "secretKey": "baz"
  }
}
```

or in YAML like:

```yaml
google: 
  accessKey: "foo"
  secretKey: "baz"
```

It would be a nice-to-have feature if we can store and reference these secrets in the run time with a `.` operator.

## Expected solution

We can allow the users to store a complex JSON object. Then, while querying it, we can allow them to use something like:
```js
ks.getSecret("secret_id", "google.accessKey") // resolves to "foo"
```

**Comment by sbk2k1** [comment]
So is this just a logical group of the env vars/secrets instead of just putting them in the .env file?

**Comment by rajdip-b** [comment]
Yep we need to allow the users to have an option - store the secret as a string or as a complex json object. So, we will track this thing in our prisma schema with a `isJSON` flag. and, the value will be set accordingly. So, we would be updating our prisma schema's `SecretVersion` model to have `jsonValue: JSON?` and `stringValue: String?`. Refer the prisma docs for supported datatypes on psql.
